_posts/2015-12-16-API-Blueprint-Future.md

```diff
@@ -0,0 +1,100 @@
+
+When encountering a new system I first need to understand its semantics. What is the data in it and what I can do with it. Only then I can start asking how to interact with it. This may sound trivial but it isn't. If my customers do not know what my is API offering why should they bother interacting with?
+
+I am already hinting here where the API Blueprint is heading.
+
+At the pinnacle of every API design is the semantics of its application domain. Once I understand **what** an API has to offer I can proceed forward to **how** to make an use of it.
+
+### Abstraction
+
+While I am pondering about the **what** of an API I don't want to worry about JSONs, URLs, protocols or other details. These decisions can wait after I define what the API is actually about.
+
+Abstraction plays the key role in any good software design. Making design of complex systems possible. Encapsulating behavior, decoupling elements enabling modularity.
+
+Let's look at what are the levels of abstraction when designing an API and its implementation.
+
+![][3]
+
+With MSON API Blueprint took a major step towards modeling application data.
+
+But make no mistake MSON isn't another syntax for JSON. MSON is agnostic to serialization media types. With MSON I can describe the data and defer the decision whether they will be send as JSON, XML or [HAL][4] over the wire.
+
+### Plans
+
+Here are the upcoming steps for API Blueprint.
+
+To finish the data description both URI parameters and HTTP headers are going to use the MSON syntax. The same syntax of parameters, headers and attributes allows the reuse of data models. It will also enable more powerful tools thanks to the data connection. [[link][5]]
+
+Next, API Blueprint resources swill start affording other resource's actions. This interlinking of resources will help the design of hypermedia systems. [[link][6]]
+
+The description of a resource and its action will be untied from the protocol layer. Separating the two will enable abstraction and open the support for other protocols. [[link][7]]
+
+Finally, API Blueprint will be able to provide the modeling of application automaton. Hypermedia clients should not use any information about the implementation. Yet, having the automaton description will help the design and enable unparalleled tools. [[link][8]]
+
+### Values
+
+To summarize, here are the values API Blueprint thrives for:
+
+* transparent &amp; open
+* simple
+* collaborative
+* encouraging abstraction
+* architectural style agnostic
+* protocol agnostic
+
+It takes more than developing the core values to have a great language. We are going to continue work on improvements to the user experience. This includes better parsers warnings, external assets [[link][9]] and splitting blueprint into many files [[link][10]].
+
+### Transparent and Open
+
+For the first time, we have prepared the public [API Blueprint roadmap][11]. Prioritized and capturing all the core values and usability features in one concise form.
```

Comment:
Last sentence is a fragment and inconsistent verb tenses.

"We have prioritized and captured all the core..."